modules/dhcp/providers/server/isc.rb

```diff
@@ -149,11 +148,22 @@ def loadSubnetData subnet
 
     def parse_config_for_subnets
       ret_val = []
-
-      @config.scan(/subnet\s+([\d\.]+)\s+netmask\s+([\d\.]+)/) do |match|
-        next unless managed_subnet? "#{match[0]}/#{match[1]}"
-
-        ret_val << Proxy::DHCP::Subnet.new(match[0], match[1])
+      @config.scan(/subnet\s+([\d\.]+)\s+netmask\s+([\d\.]+)\s*{([\w|;|\.|,|\-|\s]*)}/) do |match|
+        options = {}
+        if match[2]
+          match[2].split(';').each do |option|
+            case option
+              when /option\s+routers\s+[\d\.]+/
+                options[:routers] = get_ip_list_from_config_line(option)
+              when /option\s+domain\-name\-servers\s+[\d\.]+/
+                options[:domain_name_servers] = get_ip_list_from_config_line(option)
+              when /range\s+[\d\.]+\s+[\d\.]+/
+                # get IP addr range used for this subnet
+                options[:range] = option.scan(/\s*((?:\d{1,3}\.){3}\d{1,3})\s*((?:\d{1,3}\.){3}\d{1,3})\s*/).first
```

Comment:
Perhaps you want to extract this regex to a method `ip_address_range`? Maybe move the regex to a constant?
 You use almost all of it below as well on `get_ip_list_from_config_line`, and it's hard to know what it is without a comment, maybe that would help making this easier to read.